Userland/Libraries/LibGL/GLLights.cpp

```diff
@@ -19,12 +19,22 @@ void glColorMaterial(GLenum face, GLenum mode)
 
 void glLightf(GLenum light, GLenum pname, GLfloat param)
 {
-    g_gl_context->gl_lightf(light, pname, param);
+    g_gl_context->gl_light(light, pname, param, 0.0f, 0.0f, 0.0f);
 }
 
-void glLightfv(GLenum light, GLenum pname, GLfloat const* param)
+void glLightfv(GLenum light, GLenum pname, GLfloat const* params)
 {
-    g_gl_context->gl_lightfv(light, pname, param);
+    g_gl_context->gl_light(light, pname, params[0], params[1], params[2], params[3]);
```

Comment:
Isn't this an OOB read on `params` for e.g. the `GL_SPOT_*` constants, which have 1 or 3 parameters? I actually prefer dereferencing the pointers before going to `SoftwareGLContext` since it will help with the list logic, but @sunverwerth made a good point about not adding too much logic to this layer, so I'm not sure what the best approach is here.

Reply:
It is. I honestly think having single point of entry functions for these calls are insanity considering the amount of pointer shenanigans we have to deal with, not to mention it falls apart when we need to support the integer version, because we can't just pass in a `GLfloat*` with a `reinterpret_cast`. 

I think it'll be best to have these functions expanded out 1:1 in the C++ end of the context. We might need to discuss this a bit more.